src/util/observer.js

```diff
@@ -136,9 +136,24 @@ export default class Observer {
         }
 
         const handlers = this.handlers[event];
-        handlers &&
-            handlers.forEach(fn => {
-                fn(...args);
-            });
+        // run fireRegionOut handlers first
+        if (handlers) {
+            if (event == "audioprocess") {
+                handlers.forEach(fn => {
+                    if (fn.name == "fireRegionOut") {
+                        fn(...args);
```

Comment:
you shouldn't put plugin related stuff into classes that are used by core wavesurfer.

Reply:
I have another approach in mind, We can add a property `order` to the handler functions, and make the observer `handlers` a priority queue instead of an array, and sort that priority queue on the `order` and in case of equal order, the handlers get executed in the default order of attaching thme.

Reply:
@thijstriemstra When I wrote this PR I knew that it will get rejected, but I just wanted to show the general idea to solve those 2 issues, now I want your input to decide how to do that.

Reply:
your suggestion sounds ok to me. Would probably be a good idea to put this enhancement in a separate PR that only touches core code.

Reply:
@thijstriemstra Can I use a priority queue dependancy like  [this](https://www.npmjs.com/package/js-priority-queue), or should it be implemented in util and tested ?

Reply:
using a library is probably best as long as it's maintained actively and used a lot. Not a fan of the one you linked based on these criteria.